week08_pomdp/practice_pytorch.ipynb

```diff
@@ -444,12 +466,50 @@
    "source": [
     "def to_one_hot(y, n_dims=None):\n",
     "    \"\"\" Take an integer tensor and convert it to 1-hot matrix. \"\"\"\n",
-    "    y_tensor = y.to(dtype=torch.int64).view(-1, 1)\n",
+    "    y_tensor = y.to(dtype=torch.int64).contiguous().view(-1, 1)\n",
     "    n_dims = n_dims if n_dims is not None else int(torch.max(y_tensor)) + 1\n",
     "    y_one_hot = torch.zeros(y_tensor.size()[0], n_dims).scatter_(1, y_tensor, 1)\n",
     "    return y_one_hot"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Build Reproducibility function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import random \n",
+    "def set_seed(seed_value: int=42) -> None:\n",
+    "  os.environ['PYTHONHASHSEED'] = f\"{seed_value}\"\n",
```

Comment:
I doubt that this one is actually required.

Reply:
It is not a MAJOR thing, but it is essential to have stable results every time to trace your problems without relying on different weights and random sampling problems every time.